ci-operator/step-registry/ibmcloud/provision/vpc/endpoints-gateway/ibmcloud-provision-vpc-endpoints-gateway-commands.sh

```diff
@@ -29,26 +29,40 @@ function findTarget() {
 }
 
 function createEndpointGateway() {
-    local vpcID="$1" sgID="$2" subnetID="$3" targetCRN="$4" vpeGatewayName="$5" ret log cmd
+    local vpcID="$1" sgID="$2" subnetID="$3" targetCRN="$4" vpeGatewayName="$5" ret log cmd counter=0
     log=$(mktemp)
     echo "ibmcloud is endpoint-gateway-delete ${vpeGatewayName} --vpc ${vpcID} -f || true" >>"${SHARED_DIR}/ibmcloud_remove_resources_by_cli.sh"
     cmd="ibmcloud is endpoint-gateway-create --vpc ${vpcID} --sg ${sgID} --new-reserved-ip '{\"subnet\":{\"id\": \"${subnetID}\"}}' --target ${targetCRN} --name ${vpeGatewayName}"
     echo "Command: $cmd"
-    eval "$cmd" &> "${log}"; ret=$?
-    cat "${log}"
-    if [[ "$ret" != "0" ]] && grep -q "endpoint gateway already exists for this service" "${log}"; then
-        echo "The endpoint gateway already exists for this service, ignore the error..."
-        return 0
-    fi
-    waitingStatus ${vpeGatewayName};  ret=$?
-    echo "${vpeGatewayName} waiting status: ${ret}"
-    run_command "ibmcloud is endpoint-gateway ${vpeGatewayName}"
-
-    if [[ "${ret}" != 0 ]]; then
-        echo "ERROR: fail to create the endpoint gateway ${vpeGatewayName} on vpc"
-        return 1
-    fi
-    return 0
+    while [ $counter -lt 5 ]
+    do
+        sleep 10
+        counter=$(expr $counter + 1)
+        eval "$cmd" &> "${log}"; ret=$?
+        cat "${log}"
+        if [[ "$ret" != "0" ]]; then
+            if grep -q "endpoint gateway already exists for this service" "${log}"; then
+                echo "The endpoint gateway already exists for this service, ignoring the error..."
+                #have the ep gateway which target is same
+                return 0 
+            fi
+        fi   
+        
+        #check the endpoint-gateway status when created or has an existed
+        if [[ "$ret" == "0" ]]; then
+            waitingStatus ${vpeGatewayName};  ret=$?
```

Comment:
The `while` loop is for creating endpoint gateway with retires, why not move `waitingStatus` out of the while loop ?